lib/puppet/resource.rb

```diff
@@ -343,6 +343,7 @@ def initialize(type, title = nil, attributes = {})
         # set the type name to the symbolic name
         type = type.name
       end
+      @exported = false
```

Comment:
Did `@exported` default to `nil` previously, e.g. when written out in YAML? Do we need a test that it's always true/false but never `nil`?